tests/integration/components/flexberry-dropdown-test.js

```diff
@@ -0,0 +1,612 @@
+  this.set('placeholder', placeholder);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownText = $component.children('div.default.text');
+
+  // Check <dropdown>'s placeholder.
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()), placeholder);
+
+  // Change placeholder's value & check <dropdown>'s placeholder again.
+  placeholder = 'dropdown has no value';
+  this.set('placeholder', placeholder);
+  assert.strictEqual(Ember.$.trim($dropdownText.text()), placeholder);
+});
+
+test('readonly mode works properly', function(assert) {
+  assert.expect(4);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    class=class
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownMenu = $component.children('div.menu');
+
+  // Check that readonly (disabled) attribute doesn't exist yet.
+  assert.strictEqual($component.hasClass('disabled'), false, 'Component\'s hasn\'t readonly');
+
+  // Activate readonly mode & check that readonly (disabled) attribute exists now & has value equals to 'readonly'.
+  this.set('class', 'disabled');
+  assert.strictEqual($component.hasClass('disabled'), true, 'Component\'s has readonly');
+
+  // Check that readonly (disabled) attribute doesn't exist now.
+  this.set('class', '');
+  assert.strictEqual($component.hasClass('disabled'), false, 'Component\'s hasn\'t readonly');
+
+  // Check that component is disabled.
+  new Ember.RSVP.Promise((resolve, reject) => {
+    Ember.run(() => {
+      $component.click();
+    });
+
+    Ember.run(() => {
+      let animation = assert.async();
+      setTimeout(() => {
+        assert.strictEqual($dropdownMenu.hasClass('animating'), false, 'Component is not active');
+
+        animation();
+
+      }, animationDuration / 2);
+    });
+  });
+});
+
+test('needChecksOnValue mode properly', function(assert) {
+  assert.expect(3);
+
+  // Create array for testing.
+  let itemsArray = ['Caption1', 'Caption2', 'Caption3'];
+  this.set('itemsArray', itemsArray);
+
+  // Check value not exists.
+  this.set('value', null);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    value=value
+    items=itemsArray
+    needChecksOnValue=needChecksOnValue
+  }}`);
+
+  assert.strictEqual(this.get('value'), null, 'Component\'s property binded to \'value\' is equals to null');
+
+  // Change property binded to 'value' & check them.
+  this.set('needChecksOnValue', true);
+  let newValue = 'Caption4';
+  let latestLoggerErrorMessage;
+  Ember.Logger.error = function(message) {
+    latestLoggerErrorMessage = message;
+  };
+
+  // Check that errors handled properly.
+  this.set('value', newValue);
+  assert.strictEqual(Ember.typeOf(latestLoggerErrorMessage) === 'string', true, 'Check message exists');
+  assert.strictEqual(latestLoggerErrorMessage.indexOf(newValue) > 0, true, 'Invalide value exists');
+});
+
+test('itemsObject render properly', function(assert) {
```

Comment:
Unconsistent comment. Something like 'Dropdown with items represented by object renders properly' will be better.